### PR TITLE
Add extra terms to Wstat implementation

### PR DESCRIPTION
**Pull request by joleroi**
This adds

* the extra terms described here
https://heasarc.gsfc.nasa.gov/xanadu/xspec/manual/XSappendixStatistics.html
to the WStat implementation

see also https://github.com/gammapy/gammapy/issues/601

* A WStat description in the docs

TODO
* [ ] add table to docs
* [ ] handle corner cases of 0 counts (maybe other PR)

Will do in future PR

### CODE REVIEW

**Comment by joleroi** [comment]
@cdeil 
If you want to comment on the docs page about wstat, go ahead (note how I adopted your sloppy style, because it really does save time :smile: )

File: docs/stats/fit_statistics.rst [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+`XSpec <https://heasarc.gsfc.nasa.gov/xanadu/xspec/>`_
+X-ray analysis packages.
+
+.. Likelihood defined per bin -> take sum
+.. Stat = -2 log (L)
+.. Code example
+
+All functions compute per-bin statistics. If you want the summed statistics for
+all bins, call sum on the output array yourself. Here's an example for the
+`~cash` statistic::
```
Comment (cdeil) [inline docs/stats/fit_statistics.rst]: I think you have to put this for the ref to resolve?
```
`~gammapy.stats.cash`
```

File: gammapy/stats/fit_statistics.py [hunk 2]
```diff
@@ -155,18 +133,40 @@ def wstat(mu_signal, n_on, n_off, alpha):
...
+    # t_b * m_b = mu_background
+    # t_s / t_b = alpha
 
-    mu_signal = np.asanyarray(mu_signal, dtype=np.float64)
     n_on = np.asanyarray(n_on, dtype=np.float64)
     n_off = np.asanyarray(n_off, dtype=np.float64)
     alpha = np.asanyarray(alpha, dtype=np.float64)
+    mu_signal = np.asanyarray(mu_signal, dtype=np.float64)
+   
+    mu_background = _get_wstat_background(n_on, n_off, alpha, mu_signal)
```
Comment (cdeil) [inline gammapy/stats/fit_statistics.py]: Use consistent variable names from the table as much as possible?
I.e. here and in many other places "background" -> "bkg".

File: gammapy/stats/fit_statistics.py [hunk 3]
```diff
@@ -155,18 +133,40 @@ def wstat(mu_signal, n_on, n_off, alpha):
...
+
+    
+    term1 = mu_signal + (1 + alpha) * mu_background 
+    term2 = - n_on * np.log(mu_signal + alpha * mu_background)
+    term3 = - n_off * np.log(mu_background) 
+    
+    stat = 2 * (term1 + term2 + term3)
 
+    if extra_terms:
+        term = _get_wstat_extra_terms(n_on, n_off)
```
Comment (cdeil) [inline gammapy/stats/fit_statistics.py]: Maybe not put `term` temp variable here?
Seems pointless.
(I do it all the time, but only if lines get too long or if it will give simpler tracebacks because fewer computations happen on the same line. Here there's no such advantage in introducing a temp variable `term`)
Just put:
```python
stat += _get_wstat_extra_terms(n_on, n_off)
```

**Comment by cdeil** [comment]
> handle corner cases of 0 counts (maybe other PR)

I think in this case the easiest way to proceed would be:
- no detailed review of formulas, pull requests just go in
- in a week when we think the implementation, tests and docs are good, we ask Regis to review the version that's in master.

(The tests currently are crap. They should have visible numbers as input (not random number generator) and assert on outputs, for common and corner cases. So basically tests have to be re-written to a large degree.)

**Comment by joleroi** [comment]
> The tests currently are crap
I disagree but no need for discussion :smile: 

I will merge and add the handling of corner cases in the future when I need it

**Comment by joleroi** [comment]
I will merge this so @registerrier can continue with the docs on WStat (explain where the extra terms come from)
